Source/Singulink.Globalization.Currency/SortedMoneySet.cs

```diff
@@ -0,0 +1,365 @@
+    }
+
+    /// <summary>
+    /// Adds the specified values to this set.
+    /// </summary>
+    /// <remarks>
+    /// Default values that are not associated with any currency are ignored.
+    /// </remarks>
+    public void AddRange(IEnumerable<Money> values)
+    {
+        bool ensureCurrenciesInRegistry = values is not IReadOnlyMoneySet s || s.Registry != _registry;
+        AddRangeInternal(values, ensureCurrenciesInRegistry);
+    }
+
+    /// <summary>
+    /// Returns an enumerator that iterates through the values in this set.
+    /// </summary>
+    public Enumerator GetEnumerator() => new(_amountLookup);
+
+    /// <summary>
+    /// Removes the value with the given currency code.
+    /// </summary>
+    public bool Remove(string currencyCode)
+    {
+        var currency = _registry[currencyCode];
+        return _amountLookup.Remove(currency);
+    }
+
+    /// <summary>
+    /// Removes the value with the given currency.
+    /// </summary>
+    public bool Remove(Currency currency)
+    {
+        bool removed = _amountLookup.Remove(currency);
+
+        if (!removed)
+        {
+            EnsureCurrencyAllowed(currency, nameof(currency));
+        }
+
+        return removed;
+    }
+
+    /// <summary>
+    /// Removes all the values from this set that match the specified currencies.
+    /// </summary>
+    public int RemoveAll(IEnumerable<Currency> currencies)
+    {
+        int count = 0;
+
+        foreach (var currency in currencies)
+        {
+            if (Remove(currency))
+                count++;
+        }
+
+        return count;
+    }
+
+    /// <summary>
+    /// Copies the values in this set to a new immutable set that uses the same registry as this set.
+    /// </summary>
+    public ImmutableSortedMoneySet ToImmutableSet() => new ImmutableSortedMoneySet(_registry, this, false);
+
+    /// <inheritdoc cref="IReadOnlyMoneySet.TryGetAmount(Currency, out decimal)"/>
+    public bool TryGetAmount(Currency currency, out decimal amount)
+    {
+        return _amountLookup.TryGetValue(currency, out amount);
+    }
+
+    /// <inheritdoc cref="IReadOnlyMoneySet.TryGetValue(Currency, out Money)"/>
+    public bool TryGetValue(Currency currency, out Money value)
+    {
+        EnsureCurrencyAllowed(currency, nameof(currency));
+
+        if (_amountLookup.TryGetValue(currency, out decimal amount))
+        {
+            value = new Money(amount, currency);
+            return true;
+        }
+
+        value = default;
+        return false;
+    }
+
+    /// <inheritdoc cref="IReadOnlyMoneySet.TryGetValue(string, out Money)"/>
+    public bool TryGetValue(string currencyCode, out Money value)
+    {
+        var currency = _registry[currencyCode];
+        return TryGetValue(currency, out value);
+    }
+
+    public bool TryGetAmount(string currencyCode, out decimal amount)
+    {
+        var currency = _registry[currencyCode];
+        return _amountLookup.TryGetValue(currency, out amount);
+    }

```

Comment:
You have TryGetAmount, then TryGetValue, then TryGetAmount, and then TryGetValue again. Sort these in alphabetical order

Reply:
Sorted.